omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+                }
+            }
+            if(errors.toString().replaceAll("[\\[\\]]", "").length() > 0) {
+                throw new ValidationException("The values of the following inputs are missing or invalid: " + errors.toString());
+            }
+
+            identifierSource.setIdentifierType(patientIdentifierType);
+            identifierSource.setName(name.toString());
+            identifierSource.setUrl(url.toString());
+            savedIdentifierSource =  save(identifierSource);
+        }
+        else if(identifierSourceType.equals("IdentifierPool")){
+            IdentifierPool identifierSource = new IdentifierPool();
+            Object batchSize = postBody.get("batchSize");
+            Object minPoolSize = postBody.get("minPoolSize");
+            Object sourceUuid = postBody.get("sourceUuid");
+            Object sequential = postBody.get("sequential");
+            Object refillWithScheduledTask = postBody.get("refillWithScheduledTask");
+
+            if (sourceUuid == null || StringUtils.isBlank(sourceUuid.toString())) {
+                errors.add("identifier souce");
+            }
+            else{
+                IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                if(poolIdentifierSource != null){
+                    identifierSource.setSource(poolIdentifierSource);
+                }else{
+                    errors.add("identifier souce");
+                }
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                if(parseBoolean(sequential.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(sequential.toString()));
+                }
+            }
+            if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                if(parseBoolean(refillWithScheduledTask.toString()) != null) {
+                    identifierSource.setSequential(parseBoolean(refillWithScheduledTask.toString()));
+                }
+            }
+            if(errors.toString().length() > 2) {
```

Comment:
Am struggling to understand why you need to first call toString() is all you want to know is how many items are in this list

Reply:
replaced with: `if(errors.size() > 0) {}`